lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -1167,6 +1242,34 @@ def rotate_axes(xs, ys, zs, zdir):
         return xs, ys, zs
 
 
+def _set_offsets3d(col_3d, offsets, zdir='z'):
+    """
+    Set the 3d offsets for the collection.
+
+    Parameters
+    ----------
+    offsets : (N, 3) or (3,) array-like
+        The offsets to be set.
+    zdir : {'x', 'y', 'z'}, default: 'z'
+        The axis in which to place the offsets.
+        See `.get_dir_vector` for a description of the values.
+    """
+    offsets = np.asanyarray(offsets)
+    if offsets.shape == (3,):  # Broadcast (3,) -> (1, 3) but nothing else.
+        offsets = offsets[None, :]
+    xs = np.asanyarray(col_3d.convert_xunits(offsets[:, 0]), float)
+    ys = np.asanyarray(col_3d.convert_yunits(offsets[:, 1]), float)
+    zs = np.asanyarray(col_3d.convert_yunits(offsets[:, 2]), float)
+    col_3d._offsets3d = juggle_axes(xs, ys, np.atleast_1d(zs), zdir)
+    col_3d.stale = True
+
+
+def _get_offsets3d(col3d):
+    """Return the offsets for the collection."""
+    # Default to zeros in the no-offset (None) case
+    return np.zeros((1, 3)) if col3d._offsets3d is None else col3d._offsets3d
```

Comment:
AFAICS, `col3d._offsets3d` can never be `None` because the only place where it is set is in `_set_offsets3d()`:

```
col_3d._offsets3d = juggle_axes(xs, ys, np.atleast_1d(zs), zdir)
```
and `juggle_axes` cannot return None.

Reply:
Yes, I think you're right